cultivation/scripts/running/parse_run_files.py

```
@@ -29,141 +60,354 @@ def parse_fit_with_fitdecode(file_path):
+                total_walk_dist = sum([s['dist_km'] for s in segments if isinstance(s['dist_km'], (int, float)) and not math.isnan(s['dist_km'])])
+                # Assertion 1: walk_time + run_time ≈ session_time (±3s)
+                print(f"[DEBUG] total_session_s: {total_session_s}")
+                print(f"[DEBUG] total_walk_time: {total_walk_time}")
+                print(f"[DEBUG] run_time: {run_time}")
+                print(f"[DEBUG] total_walk_time + run_time: {total_walk_time + run_time}")
+                print(f"[DEBUG] Difference: {total_session_s - (total_walk_time + run_time)}")
+                # --- Deep dive into time sums and row counts ---
+                if 'dt' in gpx_df:
+                    print(f"[DEBUG] Sum of gpx_df['dt']: {gpx_df['dt'].sum()}")
+                if 'is_walk' in gpx_df and 'dt' in gpx_df:
+                    print(f"[DEBUG] Sum of dt where walk: {gpx_df.loc[gpx_df['is_walk'] == True, 'dt'].sum()}")
```

Ruff (0.8.2)

Avoid equality comparisons to `True`; use `if gpx_df['is_walk']:` for truth checks

Replace with `gpx_df['is_walk']`

(E712)